main.tf

```diff
@@ -110,3 +172,156 @@ resource "aws_ecs_capacity_provider" "this" {
 
   tags = merge(var.tags, try(each.value.tags, {}))
 }
+
+################################################################################
+# Task Execution - IAM Role
+# https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_execution_IAM_role.html
+################################################################################
+
+locals {
+  task_exec_iam_role_name = try(coalesce(var.task_exec_iam_role_name, var.cluster_name), "")
+
+  create_task_exec_iam_role = var.create && var.create_task_exec_iam_role
+  create_task_exec_policy   = local.create_task_exec_iam_role && var.create_task_exec_policy
+}
+
+data "aws_iam_policy_document" "task_exec_assume" {
+  count = local.create_task_exec_iam_role ? 1 : 0
+
+  statement {
+    sid     = "ECSTaskExecutionAssumeRole"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["ecs-tasks.${data.aws_partition.current.dns_suffix}"]
+    }
+  }
+}
+
+resource "aws_iam_role" "task_exec" {
+  count = local.create_task_exec_iam_role ? 1 : 0
+
+  name        = var.task_exec_iam_role_use_name_prefix ? null : local.task_exec_iam_role_name
+  name_prefix = var.task_exec_iam_role_use_name_prefix ? "${local.task_exec_iam_role_name}-" : null
+  path        = var.task_exec_iam_role_path
+  description = coalesce(var.task_exec_iam_role_description, "Task execution role for ${var.cluster_name}")
+
+  assume_role_policy    = data.aws_iam_policy_document.task_exec_assume[0].json
+  permissions_boundary  = var.task_exec_iam_role_permissions_boundary
+  force_detach_policies = true
+
+  tags = merge(var.tags, var.task_exec_iam_role_tags)
+}
+
+resource "aws_iam_role_policy_attachment" "task_exec_additional" {
+  for_each = { for k, v in var.task_exec_iam_role_policies : k => v if local.create_task_exec_iam_role }
+
+  role       = aws_iam_role.task_exec[0].name
+  policy_arn = each.value
+}
+
+data "aws_iam_policy_document" "task_exec" {
+  count = local.create_task_exec_policy ? 1 : 0
+
+  # Pulled from AmazonECSTaskExecutionRolePolicy
```

Comment:
we could, yes - but this policy has not changed for several years, users in regulated industries generally want to avoid managed policies since the permissions can change at any time without them knowing, and lastly - it simplifies things to just one policy attached to the role